plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/editors/data/preferences/PrefPageResultSetPresentation.java

```diff
@@ -88,7 +91,11 @@ protected Control createPreferenceContent(@NotNull Composite parent) {
             ((GridData)uiGroup.getLayoutData()).horizontalSpan = 2;
             autoSwitchMode = UIUtils.createCheckbox(uiGroup, DataEditorsMessages.pref_page_database_resultsets_label_switch_mode_on_rows, false);
             showDescription = UIUtils.createCheckbox(uiGroup, DataEditorsMessages.pref_page_database_resultsets_label_show_column_description, false);
-            columnWidthByValue = UIUtils.createCheckbox(uiGroup, DataEditorsMessages.pref_page_database_resultsets_label_calc_column_width_by_values, DataEditorsMessages.pref_page_database_resultsets_label_calc_column_width_by_values_tip, false, 1);
+            UIUtils.createControlLabel(uiGroup, DataEditorsMessages.pref_page_database_resultsets_label_calc_column_width_method);
+            columnWidthCalcCombo = new Combo(uiGroup, SWT.BORDER | SWT.DROP_DOWN | SWT.READ_ONLY);
+            columnWidthCalcCombo.add(DataEditorsMessages.pref_page_database_resultsets_label_calc_column_width_title_and_values);
+            columnWidthCalcCombo.add(DataEditorsMessages.pref_page_database_resultsets_label_calc_column_width_values);
+            columnWidthCalcCombo.add(DataEditorsMessages.pref_page_database_resultsets_label_calc_column_width_title_description_values);
```

Comment:
Here you can iterate over the enum's values and use its name as the label.

For example:

```java
public enum GridColumnCalcWidthMethod {
    TITLE_AND_VALUES("Title and values"),
    VALUES("Only values"),
    TITLE_DESCRIPTION_VALUES("Title, description, and values");

    private final String label;

    GridColumnCalcWidthMethod(@NotNull String label) {
        this.label = label;
    }

    @NotNull
    public String getLabel() {
        return label;
    }
}
```

And then populate the combo like so:

```java
for (GridColumnCalcWidthMethod method : GridColumnCalcWidthMethod.values()) {
    columnWidthCalcCombo.add(method.getLabel());
}
```